Validation/HGCalValidation/plugins/HGCalWaferValidation.cc

```diff
@@ -213,36 +233,54 @@ void HGCalWaferValidation::ProcessWaferLayer(DDCompactView::GraphWalker& walker)
       //edm::LogVerbatim(logcat) << "rotStr " << rotStr << " rotCode " << rotCode;
 
       // convert shape code to wafer types defined in HGCalTypes.h
-      waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferOut;
-      if (shapeStr == "F")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferFull;
-      else if (shapeStr == "a")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferHalf;
-      else if (shapeStr == "am")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferHalf2;
-      else if (shapeStr == "b")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferFive;
-      else if (shapeStr == "bm")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferFive2;
-      else if (shapeStr == "c")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferThree;
-      else if (shapeStr == "d")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferSemi;
-      else if (shapeStr == "dm")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferSemi2;
-      else if (shapeStr == "g")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferChopTwo;
-      else if (shapeStr == "gm")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferChopTwoM;
+      waferInfo.shapeCode = shapeStr == "F"    ? HGCalTypes::WaferPartialType::WaferFull
+                            : shapeStr == "a"  ? HGCalTypes::WaferPartialType::WaferHalf
+                            : shapeStr == "am" ? HGCalTypes::WaferPartialType::WaferHalf2
+                            : shapeStr == "b"  ? HGCalTypes::WaferPartialType::WaferFive
+                            : shapeStr == "bm" ? HGCalTypes::WaferPartialType::WaferFive2
+                            : shapeStr == "c"  ? HGCalTypes::WaferPartialType::WaferThree
+                            : shapeStr == "d"  ? HGCalTypes::WaferPartialType::WaferSemi
+                            : shapeStr == "dm" ? HGCalTypes::WaferPartialType::WaferSemi2
+                            : shapeStr == "g"  ? HGCalTypes::WaferPartialType::WaferChopTwo
+                            : shapeStr == "gm" ? HGCalTypes::WaferPartialType::WaferChopTwoM
+                                               : HGCalTypes::WaferPartialType::WaferOut;
 
       waferInfo.rotCode = rotCode;
       // populate the map
-      waferData[waferCoord] = waferInfo;
-      waferValidated[waferCoord] = false;
+      waferData_[waferCoord] = waferInfo;
+      waferValidated_[waferCoord] = false;
     }
   } while (walker.nextSibling());
 }
 
+// ------------ check if wafer thickness in geo matches in file (true = is a match) ------------
+bool HGCalWaferValidation::isThicknessMatched(const int geoThickClass, const int fileThickness) {
+  if (geoThickClass == 0 && fileThickness == 120)
+    return true;
+  if (geoThickClass == 1 && fileThickness == 200)
+    return true;
```

Comment:
aren't these ifs else if ?

Reply:
Given the `return` statements, `else if` would act exactly the same